src/Compilers/CSharp/Portable/FlowAnalysis/FlowAnalysisPass.cs

```diff
@@ -29,28 +29,41 @@ internal class FlowAnalysisPass
         public static BoundBlock Rewrite(
             MethodSymbol method,
             BoundBlock block,
-            DiagnosticBag diagnostics,
+            TypeCompilationState compilationState,
+            BindingDiagnosticBag diagnostics,
             bool hasTrailingExpression,
             bool originalBodyNested)
         {
 #if DEBUG
             // We should only see a trailingExpression if we're in a Script initializer.
             Debug.Assert(!hasTrailingExpression || method.IsScriptInitializer);
-            var initialDiagnosticCount = diagnostics.ToReadOnly().Length;
+            var initialDiagnosticCount = diagnostics.ToReadOnly().Diagnostics.Length;
```

Comment:
The change from `DiagnosticBag.ToReadOnly()` to `BindingDiagnosticBag.ToReadOnly()` causes more allocations (will allocate for diagnostics and for dependencies).
Seems like we could avoid allocating for dependencies (since we don't need them at all), or even better we could avoid allocating entirely (since we only need the count of diagnostics, not the diagnostics themselves).

Reply:
I'm not strongly opposed to making a change here, but this is in a debug-only section, FWIW.